django/core/checks/urls.py

```diff
@@ -28,6 +28,41 @@ def check_resolver(resolver):
         return []
 
 
+@register(Tags.urls)
+def check_url_namespaces_unique(app_configs, **kwargs):
+    """
+    Checks that all URL namespaces used in applications are unique.
+    """
+    if not getattr(settings, 'ROOT_URLCONF', None):
+        return []
+
+    from django.urls import get_resolver
+    resolver = get_resolver()
+
+    all_namespaces = _load_all_namespaces(resolver)
+    for namespace in all_namespaces:
+        if all_namespaces.count(namespace) > 1:
```

Comment:
I think this can be made faster and clearer by using a `collections.Counter` across all the namespace names